lib/service-data/specification/validate-schema.unit.spec.js

```diff
@@ -0,0 +1,188 @@
+        valid[2].path.includes(`${dataSrc}definition.data.invalid.double.error.json`)
+
+      t.equal(correctPaths, true, 'it should report the correct paths for the reported errors')
+
+      updateFixedPaths(valid)
+
+      t.deepEqual(result, expected, 'it should report the correct errors')
+    })
+})
+
+test('When validating files that are expected to be invalid against a schema but which are not', t => {
+  t.plan(2)
+
+  const expected = getExpected('specs-expected-invalid')
+
+  validateSchema('definition.data', {specs, invalid: validFiles})
+    .then(result => {
+      const invalid = result.invalid
+      const dataSrc = 'data/service-data/specification/specs-a/specifications/definition/data/data/valid/'
+      const correctPaths =
+        invalid[0].path.includes(`${dataSrc}definition.data.json`) &&
+        invalid[1].path.includes(`${dataSrc}definition.data.optional.json`)
+
+      t.equal(correctPaths, true, 'it should report the correct paths for the reported errors')
+
+      updateFixedPaths(invalid)
+
+      t.deepEqual(result, expected, 'it should report the correct errors')
+    })
+})
+
+test('When validating against a schema and requesting debug information', t => {
+  t.plan(1)
+
+  const inspect = stdout.inspect()
+  validateSchema('definition.data', {specs, debug: true})
+    .then(result => {
+      inspect.restore()
+      const consoleOutput = inspect.output[0].trim()
+      const expectedOutput = consoleOutput.startsWith('{\n  "files": [\n    [\n')
+      t.equal(expectedOutput, true, 'it should output the debug information about which files will be processed')
+    })
+})
+
+test('When validating against a schema and requesting debug information but warning is enabled', t => {
+  t.plan(1)
+
+  const inspect = stdout.inspect()
+  validateSchema('definition.data', {specs, debug: true, warn: true})
+    .then(result => {
+      inspect.restore()
+      const consoleOutput = inspect.output[0]
+      t.equal(consoleOutput, undefined, 'it should not output the debug information')
+    })
+})
+
+test('When validating against a schema and encountering more than one error', t => {
+  t.plan(2)
+  const doubleError = [`${definitionInvalid}/definition.data.invalid.double.error.json`]
+
+  validateSchema('definition.data', {specs, valid: doubleError})
+    .then(result => {
+      const expected = getExpected('specs-expected-double.error')
+      updateFixedPaths(result.valid)
+      t.deepEqual(result, expected, 'it should report all the errors encountered')
+    })
+
+  validateSchema('definition.data', {specs, valid: doubleError, allErrors: false})
+    .then(result => {
+      const expected = getExpected('specs-expected-double.error.show-first')
+      updateFixedPaths(result.valid)
+      t.deepEqual(result, expected, 'it should only report the first error encountered')
+    })
+})
+
+// test('When validating a file that determines its schema automatially', t => {
```

Comment:
can you delete this or implement the test?

Reply:
Yep! I'll delete (it's one of AR's, so it's probably been commented out for a while)

Reply:
Done